visualizations/store-map-viz/components/Regions.tsx

```diff
@@ -13,23 +13,33 @@ import countries from "../geo/countries.geojson.json";
 import geoUSStates from "../geo/us-states/us-states";
 import allUKRegions from "../geo/uk-regions/all-uk-regions";
 
+import  Gradient  from "javascript-color-gradient";
+
 const Regions = () => {
-  const { accountId, regionsQuery, fetchInterval, ignorePicker, defaultSince } =
-    useProps();
+  const { accountId, regionsQuery, fetchInterval, ignorePicker, defaultSince, heatMapSteps, regionColors } = useProps();
 
   const [regions, setRegions] = useState([]);
+  const [HMSteps, setHMSteps] = useState(0);
+  const [HMColors,setHMColors] = useState([]);
+  const [customColors,setCustomColors] = useState(null);
+
 
-  const defSinceString =
-    defaultSince === undefined || defaultSince === null
-      ? ""
-      : " " + defaultSince;
-  if (regionsQuery === null || regionsQuery === undefined) {
-    return null;
-  }
 
   const { timeRange } = useContext(PlatformStateContext);
 
   useEffect(() => {
+
+    const defSinceString =
+    defaultSince === undefined || defaultSince === null
+      ? ""
+      : " " + defaultSince;
+    if (regionsQuery === null || regionsQuery === undefined) {
+      return null;
+    }
+
+    setHMSteps(heatMapSteps && heatMapSteps!="" ? parseInt(heatMapSteps) : 0);
```

Comment:
This state setting was all to try and force a re-render whne config options changed.  It seemed unnecessary to me but i couldnt find anbother way to trigger the re-render on prop change.